RDK-57493:Run L2 test with device mgmt native platform container

dharshini555 opened the pull request:
Reason for change: To run L2 test with device management native-platform container
Test Procedure: run L2 test cases
Risks: Medium
Priority: P1

rdkcmf-jenkins commented on containers/compose.yaml:
```diff
@@ -1,19 +1,19 @@
 
   l2-container:
-    image: "native-platform:latest"
+    image: "ghcr.io/rdkcentral/docker-device-mgt-service-test/native-platform"
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
b2a341b5a3847339f23df71f43e05f35
PRESENT
-->

# Coverity Issue - Container requests ability to craft raw network packets
The Docker container requests the `NET_RAW` capability, either explicitly or by default, granting access to the host's network interfaces.

_High Impact, CWE-269_
_SIGMA.container_requesting_net_raw_

## How to fix
 Explicitly remove the `NET_RAW` capability by adding either `NET_RAW` or `ALL` to the `cap_drop` list and avoiding the `NET_RAW` capability in the `cap_add` list.

rdkcmf-jenkins replied:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
f5541ff3839050350bcc56fb31537027
PRESENT
-->

# Coverity Issue - Unspecified software version
The Docker Compose file does not explicitly set `image` attribute with a specific stable version. Pinning the version of the base image will make the container being built more predictable. Relying on the latest version may silently inherit newer packages, which in the best case may affect application reliability and in the worst case may introduce vulnerabilities.

_Low Impact, CWE-829_
_SIGMA.unspecified_software_version_

## How to fix
 Explicitly set the `image` attribute to a specific stable version.

rdkcmf-jenkins replied:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
57e3691f802e4d264e4fe03dd22610f4
PRESENT
-->

# Coverity Issue - Least privilege violation
The Docker container is not configured to drop all default capabilities and then add only required one. Thus the container may be granted excessive privileges which violate the least privilege principle.

_Low Impact, CWE-284_
_SIGMA.least_privilege_violation_

## How to fix
 Remove all the default capabilities from the list of capabilities by setting the `cap_drop` value to `ALL` and then add only specific capabilities needed by a container.

rdkcmf-jenkins replied:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
a14f71f96a56e9b7e78549a810c22e1e
PRESENT
-->

# Coverity Issue - Container allows filesystem write
The docker service container is configured to permit writing to the root filesystem. This makes some security attack vectors such as privilege escalation, denial-of-service or authorization bypass possible since the container instance's filesystem can be tampered with.

_Low Impact, CWE-552_
_SIGMA.container_filesystem_write_

## How to fix
 Explicitly set the `read-only` attribute of the service to `true` to create a service container with a read-only filesystem.

rdkcmf-jenkins replied:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
37b342c7ba6f2184bc556db154c3be27
PRESENT
-->

# Coverity Issue - Container privilege escalation allowed
The Docker Compose file sets the `no-new-privileges` attribute to `false` or omits it as the default value is `false`. This doesn't restrict the container from acquiring additional privileges via SUID or SGID bits. This allows privilege escalation, meaning a user can set the security context of the container and can perform root-level operations.

_Low Impact, CWE-269_
_SIGMA.container_privilege_escalation_allowed_

## How to fix
 Disable container privilege escalation by explicitly setting `no-new-privileges` to `true`.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Container allows filesystem write.
The docker service container is configured to permit writing to the root filesystem. This makes some security attack vectors such as privilege escalation, denial-of-service or authorization bypass possible since the container instance's filesystem can be tampered with.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Container privilege escalation allowed.
The Docker Compose file sets the `no-new-privileges` attribute to `false` or omits it as the default value is `false`. This doesn't restrict the container from acquiring additional privileges via SUID or SGID bits. This allows privilege escalation, meaning a user can set the security context of the container and can perform root-level operations.

rdkcmf-jenkins replied:
High impact. Security issue.
Sigma. Container requests ability to craft raw network packets.
The Docker container requests the `NET_RAW` capability, either explicitly or by default, granting access to the host's network interfaces.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Least privilege violation.
The Docker container is not configured to drop all default capabilities and then add only required one. Thus the container may be granted excessive privileges which violate the least privilege principle.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Unspecified software version.
The Docker Compose file does not explicitly set `image` attribute with a specific stable version. Pinning the version of the base image will make the container being built more predictable. Relying on the latest version may silently inherit newer packages, which in the best case may affect application reliability and in the worst case may introduce vulnerabilities.

rdkcmf-jenkins commented on containers/compose.yaml:
```diff
@@ -1,19 +1,19 @@
 
   mock-xconf:
-    image: "mock-xconf:latest"
+    image: "ghcr.io/rdkcentral/docker-device-mgt-service-test/mockxconf:latest"
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
09fa6022fae9ec09423e1d37f27a6ded
PRESENT
-->

# Coverity Issue - Unspecified software version
The Docker Compose file does not explicitly set `image` attribute with a specific stable version. Pinning the version of the base image will make the container being built more predictable. Relying on the latest version may silently inherit newer packages, which in the best case may affect application reliability and in the worst case may introduce vulnerabilities.

_Low Impact, CWE-829_
_SIGMA.unspecified_software_version_

## How to fix
 Explicitly set the `image` attribute to a specific stable version.

shibu-kv replied:
This file is for community reference and not used for image generation and CI flows.

rdkcmf-jenkins commented on containers/compose.yaml:
```diff
@@ -1,19 +1,19 @@
 
   mock-xconf:
-    image: "mock-xconf:latest"
+    image: "ghcr.io/rdkcentral/docker-device-mgt-service-test/mockxconf:latest"
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
6bde69707b95fb96db66450551957cf1
PRESENT
-->

# Coverity Issue - Least privilege violation
The Docker container is not configured to drop all default capabilities and then add only required one. Thus the container may be granted excessive privileges which violate the least privilege principle.

_Low Impact, CWE-284_
_SIGMA.least_privilege_violation_

## How to fix
 Remove all the default capabilities from the list of capabilities by setting the `cap_drop` value to `ALL` and then add only specific capabilities needed by a container.

rdkcmf-jenkins replied:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
59702c5c3aa5ff800dd2088109d1168a
PRESENT
-->

# Coverity Issue - Container allows filesystem write
The docker service container is configured to permit writing to the root filesystem. This makes some security attack vectors such as privilege escalation, denial-of-service or authorization bypass possible since the container instance's filesystem can be tampered with.

_Low Impact, CWE-552_
_SIGMA.container_filesystem_write_

## How to fix
 Explicitly set the `read-only` attribute of the service to `true` to create a service container with a read-only filesystem.

rdkcmf-jenkins replied:
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
9679ade33bfb4479eba448e726cd3798
PRESENT
-->

# Coverity Issue - Container privilege escalation allowed
The Docker Compose file sets the `no-new-privileges` attribute to `false` or omits it as the default value is `false`. This doesn't restrict the container from acquiring additional privileges via SUID or SGID bits. This allows privilege escalation, meaning a user can set the security context of the container and can perform root-level operations.

_Low Impact, CWE-269_
_SIGMA.container_privilege_escalation_allowed_

## How to fix
 Disable container privilege escalation by explicitly setting `no-new-privileges` to `true`.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Container allows filesystem write.
The docker service container is configured to permit writing to the root filesystem. This makes some security attack vectors such as privilege escalation, denial-of-service or authorization bypass possible since the container instance's filesystem can be tampered with.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Container privilege escalation allowed.
The Docker Compose file sets the `no-new-privileges` attribute to `false` or omits it as the default value is `false`. This doesn't restrict the container from acquiring additional privileges via SUID or SGID bits. This allows privilege escalation, meaning a user can set the security context of the container and can perform root-level operations.

rdkcmf-jenkins replied:
High impact. Security issue.
Sigma. Container requests ability to craft raw network packets.
The Docker container requests the `NET_RAW` capability, either explicitly or by default, granting access to the host's network interfaces.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Least privilege violation.
The Docker container is not configured to drop all default capabilities and then add only required one. Thus the container may be granted excessive privileges which violate the least privilege principle.

rdkcmf-jenkins replied:
Low impact. Security issue.
Sigma. Unspecified software version.
The Docker Compose file does not explicitly set `image` attribute with a specific stable version. Pinning the version of the base image will make the container being built more predictable. Relying on the latest version may silently inherit newer packages, which in the best case may affect application reliability and in the worst case may introduce vulnerabilities.

rdkcmf-jenkins commented on containers/compose.yaml:
```diff
@@ -1,19 +1,19 @@
 
   mock-xconf:
-    image: "mock-xconf:latest"
+    image: "ghcr.io/rdkcentral/docker-device-mgt-service-test/mockxconf:latest"
```
<!-- Comment managed by coverity-report-output-v7 action, do not modify!
7a771baa3d9df06656230d9de2914fb9
PRESENT
-->

# Coverity Issue - Container requests ability to craft raw network packets
The Docker container requests the `NET_RAW` capability, either explicitly or by default, granting access to the host's network interfaces.

_High Impact, CWE-269_
_SIGMA.container_requesting_net_raw_

## How to fix
 Explicitly remove the `NET_RAW` capability by adding either `NET_RAW` or `ALL` to the `cap_drop` list and avoiding the `NET_RAW` capability in the `cap_add` list.

shibu-kv replied:
This file is for community reference and not used for image generation and CI flows.

rdkcmf-jenkins commented:
Coverity detected 10 issues; a security concern.

dharshini555 committed:
RDK-57495:Run L1 test cases

Reason for change: To run L1 mock test cases
Test Procedure: run L1 test cases
Risks: Medium
Priority: P1
Signed-off-by: PriyaDharshini_Kathiravan <priyakathiravan05@gmail.com>

dharshini555 committed:
Revert "RDK-57495:Run L1 test cases"

This reverts commit 69323e0605d34d60e3b35e18c6a947be9f4a8ff4.

dharshini555 committed:
RDK-57493:Run L2 test with device mgmt native platform container

Reason for change: To run L2 test with device management native-platform container
Test Procedure: run L2 test cases
Risks: Medium
Priority: P1

dharshini555 committed:
Update compose.yaml

dharshini555 committed:
Update L2-tests.yml

dharshini555 committed:
Update L2-tests.yml

shibu-kv committed:
RDK-57493: Reuse prebuilts containers

shibu-kv committed:
RDK-57493: Reuse prebuilts containers

shibu-kv committed:
Merge branch 'develop' into topic/RDK-57394
